src/NFT/AgentCommunication.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.22;
+
+import "@openzeppelin/contracts/utils/structs/DoubleEndedQueue.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract AgentCommunication is Ownable {
+    error MessageNotSentByAgent();
+
+    mapping(address => DoubleEndedQueue.Bytes32Deque) public queues;
+    uint256 public minimumValueForSendingMessageInWei;
+
+    event NewMessageSent(address indexed sender, bytes32 message);
+
+    constructor() Ownable(msg.sender) {
+        minimumValueForSendingMessageInWei = 10000000000000; // 0.00001 xDAI
```

Comment:
Do we want this to be hard-coded? 🤔 That means if we want to change it, we need to  re-deploy contract and agents will lose un-read messages.

Reply:
The owner can change it via `adjustMinimumValueForSendingMessage`